### DISCUSSION TITLE
ESP32CAM - add image rotation option/function

### DISCUSSION THREAD

**Discussion by nikvostro** [arendst/Tasmota#12572] [Ideas]
It would be good to add ESP32CAM image rotation option/function for Scripting language and in general. Currently we have only flip and mirror settings available.

**Comment by sfromis** [comment]
Just rotate the camera :-) 

Rotation might not work well when the image format is not square.

**Reply by nikvostro** [reply]
Oh, thanks! Great solution 🙂👍 I'm already  applied some brown permanent marker to the lens to decrease brightness and to add some retro style))) And I unfocused lenses to add some blur ))

I'm just taking about rotating for +/- 90 degrees. Is that just about simple operations with X and Y coordinates for each pixel? Regardless of image format (size).

image[y][x] /* assuming this is the original orientation */ image[x][original_width - y] /* rotated 90 degrees ccw */ image[original_height - x][y] /* 90 degrees cw */

**Comment by gemu2015** [comment]
while this is possible it would slow down frame rate extremely.
the camera chip generates jpeg images. these would have to be converted to raw rgb images, then rotated and again coded to jpeg.
